server/graphql/v1/mutations/collectives.js

```diff
@@ -631,3 +632,42 @@ export async function claimCollective(_, args, req) {
 
   return collective;
 }
+
+export async function archiveCollective(_, args, req) {
+  if (!req.remoteUser) {
+    throw new errors.Unauthorized({
+      message: 'You need to be logged in to archive a collective',
+    });
+  }
+
+  const collective = await models.Collective.findByPk(args.id);
+  if (!collective) {
+    throw new errors.NotFound({
+      message: `Collective with id ${args.id} not found`,
+    });
+  }
+
+  if (!req.remoteUser.isAdmin(collective.id)) {
+    throw new errors.Unauthorized({
+      message: 'You need to be logged in as a core contributor.',
+    });
+  }
+
+  let tiers = await collective.getTiers();
+  tiers = tiers.map(async tier => {
+    return await tier.destroy();
```

Comment:
This is confusing:

- don't transform the `tiers` variable into an array of delete promises
- the use of await doesn't make sense to me, it's inside an async function with a single instruction